internal/controller/gateway_mutator.go

```diff
@@ -389,3 +393,73 @@ func (g *gatewayMutator) mutatePod(ctx context.Context, pod *corev1.Pod, gateway
 	}
 	return nil
 }
+
+// fetchGatewayAndConfig returns the Gateway and the referenced GatewayConfig (if present).
+func (g *gatewayMutator) fetchGatewayAndConfig(ctx context.Context, gatewayName, gatewayNamespace string) (*gwapiv1.Gateway, *aigv1a1.GatewayConfig) {
```

Comment:
The Gateway is not used atm so let's simplify 
```suggestion
func (g *gatewayMutator) fetchGatewayConfig(ctx context.Context, gatewayName, gatewayNamespace string) (*aigv1a1.GatewayConfig) {
```